File: src/main/java/com/ultreon/mods/motobox/automobile/NewWheelBase.java
package com.ultreon.mods.motobox.automobile;

import io.github.foundationgames.automobility.automobile.WheelBase;

@SuppressWarnings("unused")
public class NewWheelBase {
    public static WheelBase offset(float separationLong, float separationWide, float offset) {
        return new OffsetWheelBase(separationLong, separationWide, offset);
    }

    public static WheelBase bicycle(float separationLong) {
        return new BicycleWheelBase(separationLong);
    }

    public static WheelBase bicycleOffset(float separationLong, float offset) {
        return new OffsetBicycleWheelBase(separationLong, offset);
    }
}

File: src/main/kotlin/com/ultreon/mods/motobox/automobile/NewWheelBase.kt
package com.ultreon.mods.motobox.automobile

import io.github.foundationgames.automobility.automobile.WheelBase

@Suppress("unused")
object NewWheelBase {
    fun offset(separationLong: Float, separationWide: Float, offset: Float): WheelBase {
        return OffsetWheelBase(separationLong, separationWide, offset)
    }

    fun bicycle(separationLong: Float): WheelBase {
        return BicycleWheelBase(separationLong)
    }

    fun bicycleOffset(separationLong: Float, offset: Float): WheelBase {
        return OffsetBicycleWheelBase(separationLong, offset)
    }
}